doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -392,8 +476,61 @@ stands for :n:`let @ident := fun {+ @binder} => @term in @term’`.
 
 .. index:: match ... with ...
 
-Definition by case analysis
----------------------------
+Definition by cases: match
+--------------------------
+
+.. insertgram term_match pattern_list_or2
+
+.. productionlist:: coq
+   term_match : match `case_item_list_comma` `term100_opt` with `or_opt` `eqn_list_or_opt` end
```

Comment:
This rule is buggy, `term100_opt` should be an optional `return_type` (`return_type_opt` after the current `return_type_opt` is renamed).